src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/integTests/tests/jvmsizing/validator/HeapSizeIncreaseValidator.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.integTests.tests.jvmsizing.validator;
+
+import static org.junit.Assert.assertTrue;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.AppContext;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.HeapSizeIncreaseAction;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.integTests.framework.api.IValidator;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.persistence.actions.PersistedAction;
+
+public class HeapSizeIncreaseValidator implements IValidator {
+
+  AppContext appContext;
+  long startTime;
+
+  public HeapSizeIncreaseValidator() {
+    appContext = new AppContext();
+    startTime = System.currentTimeMillis();
+  }
+
+  @Override
+  public boolean checkDbObj(Object object) {
+    if (object == null) {
+      return false;
+    }
+
+    PersistedAction persistedAction = (PersistedAction) object;
+    return checkPersistedAction(persistedAction);
+  }
+
+  private boolean checkPersistedAction(final PersistedAction persistedAction) {
```

Comment:
Can we check more action attributes, like impacted node IPs & IDs, cool off period, action name etc. (similar to [here](https://github.com/opendistro-for-elasticsearch/performance-analyzer-rca/blob/master/src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/integTests/tests/cache_tuning/validator/FieldDataCacheDeciderValidator.java#L68)) ?

Reply:
Done.